cupy/_core/_routines_statistics.pyx

```diff
@@ -438,6 +442,9 @@ cpdef ndarray _median(
 
     out = _mean(
         part[indexer], axis=axis, dtype=None, out=out, keepdims=keepdims)
+    if part.dtype.kind in 'fc':
+        isnan = _exists_nan(part, axis=axis, keepdims=keepdims)
+        out = cupy.where(isnan, numpy.nan, out)
```

Comment:
It causes slight performance degradation.
```py
import cupy
from cupy import testing
from cupyx.profiler import benchmark


shape = (64, 1024, 1024)
x1 = testing.shaped_random(shape, xp=cupy, dtype='f', seed=0, scale=100000000)
perf = benchmark(cupy.median, (x1,), {'axis': -1}, n_repeat=100)

print(perf)

x1 = x1.ravel()
perf = benchmark(cupy.median, (x1,), n_repeat=100)

print(perf)
```

- master:
```
median              :    CPU:37751.316 us   +/-31.175 (min:37590.713 / max:37942.023) us     GPU-0:37766.001 us   +/-31.199 (min:37605.377 / max:37957.630) us
median              :    CPU: 3272.639 us   +/-37.611 (min: 3127.364 / max: 3294.094) us     GPU-0: 3277.998 us   +/-37.440 (min: 3132.416 / max: 3299.328) us
```

- This PR
```
median              :    CPU:37874.557 us   +/-39.133 (min:37633.808 / max:38127.786) us     GPU-0:38012.037 us   +/-36.026 (min:37771.263 / max:38213.631) us
median              :    CPU: 3881.922 us   +/-688.615 (min: 3370.147 / max: 5657.988) us     GPU-0: 4082.176 us   +/-726.374 (min: 3547.136 / max: 5958.656) us
```

Reply:
its around 2,3% so I think it is ok!